ai/main.py

```
@@ -1,29 +1,34 @@
+        book_info = get_book(book_id, authorization)
+        story = get_all_story(book_id, authorization)
+        if book_info and story:
+            genre = book_info.get("genre")
+            character = book_info.get("character")
+            background = book_info.get("background")
+            result = ai.generate_story(genre, character, background, story, choice)
+            return {"status": "success", "result":result}
+        else:
+            raise HTTPException(status_code=404,detail="데이터가 없습니다.")
+    except Exception as e:
+        raise HTTPException(status_code=500,detail=f"책 조회 실패: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)